tutorial_pl/README.md

```diff
@@ -0,0 +1,152 @@
+	int max_n = atoi(argv[1]),
+	    max_val = atoi(argv[2]);
+
+	int n = rd(1, max_n);
+	cout << n << '\n';
+	for(int i = 0; i < n; ++i)
+		cout << rd(-max_val, max_val) << ' ';
+}
+```
+
+By ją uruchomić, należy skompilować ją normalnie i odpalić `./generatorka 5 10` (czyli argumentami do generatorki są liczby 5, a potem 10).
+
+# Generowanie testów
+
+Ustal jakie będą grupy testów, ile będzie testów w danej grupie i jakie będą ich wielkości. Przydziałem punktów do każdej grupy testów zajmiemy się później. Każdy plik wejściowy/wyjściowy zazwyczaj jest nazywany w sposób tag1a, gdzie '1' to numer grupy, a 'a' to numer testu w grupie. Testy przykładowe mają mieć numer grupy 0. Użytkownik widzi status swojego kodu na testach przykładowych.
+
+Sip potrafi automatycznie generować testy za pomocą pliku Sipfile. Przykładowy plik Sipfile:
+```
+# czas (sekund), po którym przerywa wzorcówkę podczas generowania testów:
+default_time_limit: 15
+static: [
+	# Tu umieść testy zrobione ręcznie (testy przykładowe)
+	# Syntax: <test-range>
+	tag0a
+]
+gen: [
+	# Tu umieść zasady do generowania testów
+	# Syntax: <test-range> <generator> [generator arguments]
+	tag1a-f utils/generatorka 5 10
+	tag2a utils/generatorka 100 100000
+	tag2b utils/generatorka 1000 10000
+	tag3-5a-d utils/generatorka 100000 1000000000
+]
+```
+
+Po zapisaniu pliku Sipfile, komenda `sip gen` wygeneruje wszystkie pliki *.in za pomocą generatorki, a potem wszystkie pliki *.out za pomocą wzorcówki.
+
+# Plik konfiguracyjny Simfile
+
+Wszelkie ustawienia dotyczące paczki znajdują się w pliku Simfile.
+
+Przykładowy, pełny plik Simfile:
+```
+name: Nazwa Zadania
+label: tag
+statement: doc/tag.pdf
+solutions: [prog/tag.cpp, prog/tags1.cpp, prog/tagb1.cpp]
+memory_limit: 256
+checker: check/checker.cpp
+limits: [
+	0 0.4
+
+	1a 0.4
+	1b 0.4
+	
+	2 2.0
+]
+scoring: [
+	0 0
+	1 25
+	2 75
+]
+tests_files: [
+	0 in/tag0.in out/0.out
+	1a in/tag1a.in out/tag1a.out
+	1b in/tag1b.in out/tag1b.out
+	2 in/tagb.in out/tagb.out
+]
+```
+* name: nazwa zadania
+* label: tag zadania
+* memory_limit: limit pamięci (w MB)
+* solutions: lista programów (domyślnie nie trzeba pisać tej linijki)
+* checker: ścieżka do checkerki (domyślnie nie trzeba)
+* limits: dla każdego testu (bez prefixu nazwy testu) maksymalny limit czasu w sekundach (domyślnie sam generuje)
+* scoring: dla każdej grupy testów ilość przyznawanych punktów (domyślnie daje po równo dla każdej grupy testów, tak by się sumowało do 100)
+* test_files: dla każdego testu ścieżka do pliku wejściowego/wyjściowego (to jakiś bezsens, nie opłaca się tego ruszać bo domyślnie i tak nie trzeba)
+
+Ponieważ nazwa zadania, label oraz limit pamięci są jedynymi parametrami, które są obowiązkowe, można je uzupełnić komendą (zamiast edytorem tekstu): `sip name [value]`, `sip label [value]`, `sip mem [value]`. Podobnie istnieją komendy: `sip main-sol [sol]`, `sip statement [value]`, `sip save scoring`.
```

Comment:
Do komendy `name` warto dodać, że w shellu czasem trzeba użyć cudzysłowów (i to lepiej pojedyńczych) np. 'Ala ma kota' lub '&, | i inne operacje'.
Do komendy `main-sol` i `statement` muszą być podane ścieżki do odpowiednich plików.